fw/http.c

```diff
@@ -3163,9 +3204,18 @@ __tfw_http_add_hdr_via(TfwHttpMsg *hm, int http_version, bool from_cache)
 		    g_vhost->hdr_via_len);
 
 	if (!from_cache) {
-		r = tfw_http_msg_hdr_add(hm, &rh);
-	}
-	else {
+		if (!hm->resp) {
+			/* Add via to request */
+			if (unlikely((r = tfw_http_msg_hdr_add(hm, &rh))))
+				return r;
+			return 0;
+		}
+		r = tfw_http_msg_expand_from_pool((TfwHttpResp *)hm, &rh);
+		if (unlikely(r))
+			return r;
+		r = tfw_http_msg_expand_from_pool((TfwHttpResp *)hm,
+						  &TFW_STR_STRING(S_CRLF));
+	} else {
```

Comment:
On the line

Reply:
I can't find this part in current version of code, please specify correct line number.